deployer/server.go

```diff
@@ -0,0 +1,401 @@
+	deployment.NewReleaseID = postgres.CleanUUID(deployment.NewReleaseID)
+
+	args, err := json.Marshal(deployID{ID: deployment.ID})
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+	if err := q.Enqueue(&que.Job{
+		Type: "Deployment",
+		Args: args,
+	}); err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(w).Encode(deployment); err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+}
+
+func getDeployment(id string) (*deployer.Deployment, error) {
+	d := &deployer.Deployment{}
+	query := "SELECT deployment_id, app_id, old_release_id, new_release_id, strategy, created_at FROM deployments WHERE deployment_id = $1"
+	err := db.QueryRow(query, id).Scan(&d.ID, &d.AppID, &d.OldReleaseID, &d.NewReleaseID, &d.Strategy, &d.CreatedAt)
+	d.ID = postgres.CleanUUID(d.ID)
+	d.OldReleaseID = postgres.CleanUUID(d.OldReleaseID)
+	d.NewReleaseID = postgres.CleanUUID(d.NewReleaseID)
+	if err != nil {
+		return nil, err
+	}
+	return d, nil
+}
+
+func sendDeploymentEvent(e deployer.DeploymentEvent) error {
+	if e.Status == "" {
+		e.Status = "running"
+	}
+	query := "INSERT INTO deployment_events (deployment_id, release_id, job_type, job_state, status) VALUES ($1, $2, $3, $4, $5)"
+	return db.Exec(query, e.DeploymentID, e.ReleaseID, e.JobType, e.JobState, e.Status)
+}
+
+// TODO: share with controller streamJobs
+func streamDeploymentEvents(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+	deploymentID := params.ByName("deployment_id")
+	var lastID int64
+	lastIDHeader := req.Header.Get("Last-Event-Id")
+	if lastIDHeader != "" {
+		var err error
+		lastID, err = strconv.ParseInt(lastIDHeader, 10, 64)
+		if err != nil {
+			http.Error(w, fmt.Sprintf(`invalid Last-Event-Id header "%s"`, lastIDHeader), 400)
+		}
+	}
+
+	var err error
+	defer func() {
+		if err != nil {
+			http.Error(w, err.Error(), 500)
+		}
+	}()
+
+	w.Header().Set("Content-Type", "text/event-stream; charset=utf-8")
+
+	sendKeepAlive := func() error {
+		if _, err := w.Write([]byte(":\n")); err != nil {
+			return err
+		}
+		w.(http.Flusher).Flush()
+		return nil
+	}
+
+	sendDeploymentEvent := func(e *deployer.DeploymentEvent) error {
+		if _, err := fmt.Fprintf(w, "id: %d\ndata: ", e.ID); err != nil {
+			return err
+		}
+		if err := json.NewEncoder(w).Encode(e); err != nil {
+			return err
+		}
+		if _, err := w.Write([]byte("\n")); err != nil {
+			return err
+		}
+		w.(http.Flusher).Flush()
+		return nil
+	}
+
+	connected := make(chan struct{})
+	done := make(chan struct{})
+	listenEvent := func(ev pq.ListenerEventType, listenErr error) {
+		switch ev {
+		case pq.ListenerEventConnected:
+			close(connected)
+		case pq.ListenerEventDisconnected:
+			close(done)
+		case pq.ListenerEventConnectionAttemptFailed:
+			err = listenErr
+			close(done)
+		}
+	}
+	listener := pq.NewListener(db.DSN(), 10*time.Second, time.Minute, listenEvent)
```

Comment:
Because this was implemented before the switch to pgx, I plan to merge this PR into the temp branch, then open up additional PRs that add logging and change to use pgx exclusively.